src/components/SignerPanel/SignerPanelContent.js

```diff
@@ -142,11 +139,15 @@ const NeedWeb3Content = ({ intent: { description, to }, onClose }) => (
   </React.Fragment>
 )
 
-const AddressLink = ({ to }) => (
-  <SafeLink href={`${etherscanBaseUrl}/address/${to}`} target="_blank">
-    {to}
-  </SafeLink>
-)
+const AddressLink = provideNetwork(({ network: { etherscanBaseUrl }, to }) => {
+  return etherscanBaseUrl && to ? (
+    <SafeLink href={`${etherscanBaseUrl}/address/${to}`} target="_blank">
+      {to}
+    </SafeLink>
+  ) : (
+    'this app'
```

Comment:
Will it always hold true that `to` is for "this app"?

Reply:
Mmm good point, it could be any app. I'll reword this to 'an address'.

Reply:
See 7a15d42